src/hotspot/share/c1/c1_CodeStubs.hpp

```diff
@@ -534,7 +558,14 @@ class ArrayCopyStub: public CodeStub {
   LIR_OpArrayCopy* _op;
 
  public:
-  ArrayCopyStub(LIR_OpArrayCopy* op): _op(op) { }
+  ArrayCopyStub(LIR_OpArrayCopy* op): _op(op) {
+    FrameMap* f = Compilation::current()->frame_map();
+#if defined(X86)
+    f->update_reserved_argument_area_size(5 * BytesPerWord);
+#else
+    f->update_reserved_argument_area_size(2 * BytesPerWord);
+#endif
+  }
```

Comment:
How about moving the CPU-specific value to CPU-specific code?  Maybe `c1_Defs_<cpu>.hpp`?

Reply:
Done